bool start
bool initialize
---
bool success
